nx/lib/nx/serving.ex

```diff
@@ -1300,7 +1317,7 @@ defmodule Nx.Serving do
       }
     )
 
-    :pg.join(Nx.Serving.PG, __MODULE__, List.duplicate(self(), partitions_count))
+    :pg.join(Nx.Serving.PG, __MODULE__, List.duplicate(self(), weight * partitions_count))
```

Comment:
Should we add the `max(1)` here? Also, could someone want the weight to be a float? Should we validate it is an integer?

Reply:
I added the `max(1, weight)` in `init` instead of `start_link` and added a check for whether the weight is an integer. 

While I could imagine that someone might want to use a float for it, I don't think it works well with the current way of letting the process join multiple times, since the "join count" is a discrete value. The floats could be rescaled approximately to an discrete "join count" (doesn't seem very elegant to me, since it would lead to big PID list, that is later sampled from), but since even with integer weights a desired float value can be approximated by a weight ratio, the current solution seems to allow enough flexibility for me. For example, if I want a node to receive 45% of all requests on average, I can just set `distribution_weight: 9` and, assuming there is one other node, the other distribution weight to `11`.